tests/llm/test_example_selector.py

```diff
@@ -0,0 +1,222 @@
+import asyncio
+from typing import Any, Dict, List
+
+import pytest
+import numpy as np
+from langchain_core.example_selectors.base import BaseExampleSelector
+from langchain_core.example_selectors import LengthBasedExampleSelector
+from langchain_core.prompts import FewShotPromptTemplate, PromptTemplate
+from langchain_core.messages import AIMessage, HumanMessage
+from pydantic import BaseModel, Field, RootModel
+
+from chatsky import MISC, RESPONSE
+from chatsky.llm.example_selector import Example, StaticExampleSelector, to_langchain_context
+from chatsky.responses.llm import LLMResponse
+
+
+class ExamplePrompt(BaseModel, arbitrary_types_allowed=True):
+    examples: StaticExampleSelector | BaseExampleSelector
+
+
+class ResponseModel(BaseModel):
+    sum: float = Field(description="Sum of the numbers")
+    prod: float = Field(description="Product of the numbers")
+
+
+class RequestModel(BaseModel):
+    operand_1: float = Field(description="The first operand")
+    operand_2: float = Field(description="The second operand")
+
+
+class MockCustomExampleSelector(BaseExampleSelector, RootModel):
+
+    root: List[Example]
+
+    def add_example(self, example: Dict[str, str]) -> Any:
+        self.root.append(Example.model_validate(example))
+
+    def unpack_model(self, example_part: str | BaseModel) -> str:
+        return str(example_part.model_dump_json()) if isinstance(example_part, BaseModel) else example_part
+
+    def select_examples(self, input_variables: Dict[str, str]) -> List[dict]:
+        subset = np.random.choice(self.root, input_variables["size"], input_variables["replace"])
+        return [
+            {"input": self.unpack_model(example.input), "output": self.unpack_model(example.output)}
+            for example in subset
+        ]
+
+
+### Tests
+
+
+@pytest.fixture(scope="function")
+def static_selector_factory():
+    def static_selector(examples: List[Example]):
+        return StaticExampleSelector(examples)
+
+    return static_selector
+
+
+@pytest.fixture(scope="function")
+def node_factory():
+    def setted_node(use_static_selector: bool):
+
+        np.random.seed(0)
+        node = {RESPONSE: LLMResponse(llm_model_name="my_model", prompt="Add numbers and return an answer.")}
+        test_examples = [
+            {"input": '{"operand_1":3.0,"operand_2":4.0}', "output": '{"sum":7.0,"prod":12.0}'},
+            {"input": RequestModel(operand_1=5.0, operand_2=6.0), "output": '{"sum":11.0,"prod":30.0}'},
+            {"input": '{"operand_1":0.0,"operand_2":8.0}', "output": ResponseModel(sum=8.0, prod=0.0)},
+            {"input": RequestModel(operand_1=-1.0, operand_2=-2.0), "output": ResponseModel(sum=-3.0, prod=2.0)},
+        ]
+
+        if use_static_selector == True:
```

Comment:
Just an FYI:

There are three main ways to compare with True:
1. `if x is True`;
2. `if x == True`;
3. `if x`.

They are all different and not knowing the difference can lead to hard-to-debug bugs.

`if x is True` checks if the object `x` is exactly the same object (as in occupies the same memory space) as the `True` object.
`if x == True` checks if `x.__eq__(True)` is true. Examples: `1 == True`,  `2 != True`, `[1] != True`, `1.0 == True`.
`if x` checks if `x.__bool__()` is true. Examples: `bool(2) is True`, `bool(0) is False`, `bool([0]) is True`, `bool([]) is False`.